staging/src/k8s.io/cloud-provider/controllers/route/route_controller.go

```diff
@@ -57,6 +56,8 @@ var updateNetworkConditionBackoff = wait.Backoff{
 }
 
 type RouteController struct {
+	MaxConcurrentRouteCreations int
```

Comment:
Modifying the field for RouteController is probably not the right approach here. 

Not 100% sure yet, but it _might_ be better to add new config type for route controller with flags, similar to service controller https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/service/config/types.go#L19-L25

Reply:
Alternatively we may want to add methods to `Routes` interface in the cloud provider to tweak this but there hasn't been much precedent in the past for adding rate limiting knobs there.

Reply:
> Modifying the field for RouteController is probably not the right approach here as this struct is not easily accessible when vendoring the cloud provider lib.

It is? The `New` function returns a `*RouteController` I don't see how that wouldn't be accessible by cloud providers vendoring this lib.
Obviously I took the current approach because it was the easiest without changing any apis and maintaining backwards compatibility for a rather unimportant (in general not to me of course :)) knob.

That being said I'm open to add a new config type for the RouteController if that is the preferred way.

I also don't like changing the `Routes` interface for this.

Reply:
>It is? The New function returns a *RouteController. I don't see how that wouldn't be accessible by cloud providers vendoring this lib.

The type is accessible, but typically you don't import this controller directly, you import `cmd/cloud-controller-manager` which will call New. There's no existing hooks to modify the instance of RouteController returned by New. 

> That being said I'm open to add a new config type for the RouteController if that is the preferred way.

I think this might be the best approach.